libcxx/include/fstream

```diff
@@ -276,6 +276,30 @@ private:
   state_type __st_;
   state_type __st_last_;
   ios_base::openmode __om_;
+  // Used to track the currently used mode and track whether the output should
```

Comment:
Review note: It might be possible to use other fields to achieve the same goal. Unfortunately there's no implementation documentation for this class. This changes seems the simplest and the easiest to validate.